client/state/jetpack-checkout/reducer.ts

```diff
@@ -21,14 +21,17 @@ export const requestStatus = keyedReducer( 'receiptId', ( state, { type } ) => {
 	return state;
 } );
 
-export const submittedSiteUrl = keyedReducer( 'receiptId', ( state, { type, payload } ) => {
-	switch ( type ) {
-		case JETPACK_CHECKOUT_UPDATE_SUPPORT_TICKET_RECEIVE:
-			return payload;
-	}
+export const submittedSiteUrl = keyedReducer(
+	'jetpackTemporarySiteId',
```

Comment:
Is not guaranteed that the receipt ID will be available post-purchase so it's better to use the site ID of the temporary site.